packages/ai-core/src/common/token-usage-service.ts

```diff
@@ -0,0 +1,46 @@
+// *****************************************************************************
+// Copyright (C) 2025 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { TokenUsageServiceClient } from './protocol';
+
+export const TokenUsageService = Symbol('TokenUsageService');
+
+export interface TokenUsage {
+    /** The input token count */
+    inputTokens: number;
+    /** The output token count */
+    outputTokens: number;
+    /** The model identifier */
+    model: string;
+    /** The timestamp of when the tokens were used */
+    timestamp: Date;
+}
+
+export interface TokenUsageService {
+    /**
+     * Records token usage for a model interaction.
+     *
+     * @param inputTokens The number of input tokens used
+     * @param outputTokens The number of output tokens generated
+     * @param model The identifier of the model that was used
+     * @returns A promise that resolves when the token usage has been recorded
+     */
+    recordTokenUsage(inputTokens: number, outputTokens: number, model: string): Promise<void>;
```

Comment:
Nitpick: To me, it seems more logical to put the model id as a first parameter and the second parameter is an object with input and output tokens. This way we can more easily extend what we track besides input and output tokens (thinking in the direction of image and audio).